### PR TITLE
upgrade docs to vuepress v2.0.0-rc and implement dark mode

### PR DESCRIPTION
**Pull request by tilalx**
### Design Overview Update

An updated design image has been incorporated to improve the visual representation of the project, now featuring Dark Mode support in the documentation. This enhancement not only aligns with contemporary aesthetic preferences but also offers an alternative viewing mode that is considerate of varying user needs and contexts. The revised design can be viewed here:

Updated Design

### Environment Specifications

The project has been thoroughly tested with the following environment setup to ensure compatibility and stability:

- Node.js version: `18`
- Yarn version: `4`
- vuepress: `2.0 rc`

### CODE REVIEW

**Comment by tilalx** [comment]
Hi @jc21,

Could you please guide me on how to access the Jenkins logs for the failed build? Also, I am wondering if Node.js version 18 is available on the Jenkins server. If it is, could we possibly select it using nvm use 18 for the build process?

Thank you!

File: docs/guide/README.md
Comment (jc21) [inline docs/guide/README.md:1]: This symlink should not be removed

**Comment by jc21** [comment]
@tilalx the node version used in CI for docs building is pretty much the latest, as it's using `node:alpine` docker image. At this time that version is `v21.5.0`

**Comment by jc21** [comment]
Can you fix up the yarn.lock file after rebasing and CI will try again.

**Comment by jc21** [comment]
The CI nodes have been updated to node 20. The preview documentation for the `develop` branch is available here:

https://develop.nginxproxymanager.com/

I noticed that the menu has a folder path in it, can we fix that in a new PR?

Screenshot 2024-01-10 at 9 43 52 am

**Comment by tilalx** [comment]
Yes, we can open a new PR and fix this small bug.
image
image

**Comment by tilalx** [comment]
Thank you as well!

**Pull request merged**